compiler/sem.nim

```diff
@@ -165,6 +165,27 @@ proc commonType*(x, y: PType): PType =
         result = newType(k, r.owner)
         result.addSonSkipIntLit(r)
 
+proc procSymInCallExpr(n: PNode): PSym =
+  assert(n.kind in nkCallKinds)
+  var it = n
+  while it.kind != nkSym:
```

Comment:
That is not correct and not necessary!  Should produce a crashing loop for ``f()()``. Proc calls are not static in Nim, they can always go through a procvar!

Reply:
Thanks for clarification, now it is clear that dotExprs that I was seeing were calls via function pointers and should be ignored.
procSymInCallExpr is removed.